packages/studio/src/components/AddPageButton/EntityIdField.tsx

```diff
@@ -0,0 +1,125 @@
+import Select, { MultiValue } from "react-select";
+import useStudioStore from "../../store/useStudioStore";
+import StreamScopeFieldLabel from "./StreamScopeFieldLabel";
+import { ChangeEvent, useCallback, useMemo, useState } from "react";
+import classNames from "classnames";
+import getSelectCssClasses from "../../utils/getSelectCssClasses";
+
+interface Props {
+  disabled: boolean;
+  updateSelection: (selectedIds: string[]) => void;
+  selectedIds?: string[];
+}
+
+export default function EntityIdField({
+  disabled,
+  updateSelection,
+  selectedIds,
+}: Props) {
+  const [entitiesRecord, fetchEntities] = useStudioStore((store) => [
+    store.accountContent.entitiesRecord,
+    store.accountContent.fetchEntities,
+  ]);
+  const availableEntityTypes = Object.entries(entitiesRecord)
+    .filter(([_entityType, entityData]) => {
+      return Object.values(entityData).length > 0;
+    })
+    .map((entry) => entry[0]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [entityType, setEntityType] = useState<string | undefined>(
+    availableEntityTypes[0]
+  );
+  const entityOptions = useEntityOptions(entityType);
+  const selectedOptions = useMemo(
+    () => selectedIds?.map((id) => ({ value: id, label: id })),
+    [selectedIds]
+  );
+
+  const onEntityTypeChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setEntityType(e.target.value);
+    },
+    []
+  );
+
+  const onEntityIdChange = useCallback(
+    (selectedItems: MultiValue<{ value: string }>) => {
+      updateSelection(selectedItems.map((item) => item.value));
+    },
+    [updateSelection]
+  );
+
+  const onMenuScrollToBottom = useCallback(async () => {
+    if (!entityType) {
+      return;
+    }
+    const currentNumEntities = Object.keys(entitiesRecord[entityType]).length;
+    if (currentNumEntities >= entitiesRecord[entityType].totalCount) {
+      return;
+    }
+    setIsLoading(true);
+    await fetchEntities(entityType, Math.floor(currentNumEntities / 50));
+    setIsLoading(false);
+  }, [entitiesRecord, entityType, fetchEntities]);
+
+  const className = classNames("flex flex-col mb-1");
```

Comment:
nit: can this variable be a little more descriptive of what element it's styling?